library/test/src/formatters/terse.rs

```diff
@@ -65,15 +69,21 @@ impl<T: Write> TerseFormatter<T> {
         color: term::color::Color,
     ) -> io::Result<()> {
         self.write_pretty(result, color)?;
-        if self.test_count % QUIET_MODE_MAX_COLUMN == QUIET_MODE_MAX_COLUMN - 1 {
+        self.test_column += 1;
+        if self.test_column == QUIET_MODE_MAX_COLUMN {
             // We insert a new line regularly in order to flush the
             // screen when dealing with line-buffered output (e.g., piping to
             // `stamp` in the rust CI).
-            let out = format!(" {}/{}\n", self.test_count + 1, self.total_test_count);
-            self.write_plain(out)?;
+            self.write_progress()?;
         }
 
-        self.test_count += 1;
+        Ok(())
+    }
+
+    fn write_progress(&mut self) -> io::Result<()> {
+        let out = format!(" {}/{}\n", self.test_column, self.total_test_count);
```

Comment:
We still need a `test_count` instead of using the column here, or else we're missing overall progress:

```
........................................................................................ 88/1000
........................................................................................ 88/1000
........................................................................................ 88/1000
...
```

Reply:
good catch, thank you! i added a test which catches this bug and also a couple others i ran into while fixing it.